script/cli/check_worker.lua

```diff
@@ -17,6 +17,96 @@ require 'vm'
 
 local export = {}
 
+local colors
+
+if not os.getenv('NO_COLOR') then
+    colors = {
+        red = '\27[31m',
+        green = '\27[32m',
+        yellow = '\27[33m',
+        blue = '\27[34m',
+        magenta = '\27[35m',
+        grey = '\27[90m',
+        reset = '\27[0m'
+    }
+else
+    colors = {
+        red = '',
+        green = '',
+        yellow = '',
+        blue = '',
+        magenta = '',
+        grey = '',
+        reset = ''
+    }
+end
+
+local severity_colors = {
+  [1] = colors.red,
+  [2] = colors.yellow,
+  [3] = colors.blue,
+  [4] = colors.green,
+}
+
+local severity_str = {
+  [1] = 'error',
+  [2] = 'warning',
+  [3] = 'info',
+  [4] = 'hint'
+}
+
+---@param path string
+---@return string
+local function relpath(path)
+    local pwd = os.getenv('PWD')
+    if pwd and path:sub(1, #pwd) == pwd then
+        path = path:sub(#pwd + 2)
+    end
+    return path
+end
+
+local function report_pretty(results)
+    for f, diags in pairs(results) do
+        local path = relpath(f:match('^file://(.+)$') or f)
+
+        local lines = {} --- @type string[]
+        pcall(function()
+            for line in io.lines(path) do
```

Comment:
(just a side note)

without the `furi.decode` fix, the path is in **invalid** format on windows
(like `/c%3A/Users/TomLau/test/test.lua`)
so this logic cannot correct open the file

with the `furi.decode` fix, now it works on windows as well 🎉